src/libs/actions/Task.js

```diff
@@ -259,10 +265,39 @@ function completeTask(taskReportID, taskTitle) {
         },
     ];
 
-    // Update optimistic data for parent report action
-    const optimisticParentReportData = ReportUtils.getOptimisticDataForParentReportAction(taskReportID, completedTaskReportAction.created, CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD);
-    if (!_.isEmpty(optimisticParentReportData)) {
-        optimisticData.push(optimisticParentReportData);
+    // Update optimistic data and failure data for parent report action
+    if (taskReport.parentReportActionID) {
+        let parentReportAction = ReportActionsUtils.getParentReportAction(taskReport);
+        if (parentReportAction && parentReportAction.reportActionID) {
+            const optimisticParentReportActionData = ReportUtils.updateOptimisticParentReportAction(
+                parentReportAction,
+                completedTaskReportAction.created,
+                CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD,
+            );
+            optimisticData.push({
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${taskReport.parentReportID}`,
+                value: {
+                    [parentReportAction.reportActionID]: optimisticParentReportActionData,
+                },
+            });
+
+            // Get assignee report from participantAccountIDs as in creating task according to https://github.com/Expensify/App/issues/23920#issuecomment-1663092717
+            const assigneeChatReportID = lodashGet(ReportUtils.getChatByParticipants(taskReport.participantAccountIDs), 'reportID');
+            // We don't need to create optmistic report action for assignee report if assignee and shareDestination are the same
+            if (assigneeChatReportID && `${assigneeChatReportID}` !== `${taskReport.parentReportID}`) {
+                parentReportAction = ReportActionsUtils.getParentReportActionForTask(taskReportID, assigneeChatReportID);
```

Comment:
Can we rename this to clonedParentReportAction? or anything that tells you this parent is a copy

Reply:
okay